test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtraction/StreamExtensionsTests.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+using Moq;
+using NuGet.Common;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Packaging.Test.PackageExtraction
+{
+    class TestableProxy : NuGet.Packaging.StreamExtensions.Testable
+    {
+        public bool MmapCopyWasCalled { get; set; }
+        public bool FileStreamCopyWasCalled { get; set; }
+        internal TestableProxy(IEnvironmentVariableReader environmentVariableReader) : base(environmentVariableReader)
+        {
+        }
+
+        internal override void MmapCopy(Stream inputStream, string fileFullPath, long size)
+        {
+            MmapCopyWasCalled = true;
+            base.MmapCopy(inputStream, fileFullPath, size);
```

Comment:
given that this is used in tests only for checking the right method was called, it's not used to validate contents, I don't think there's a benefit to calling `base.MmapCopy`.

By coincidence, I think that validating contents is useful, because I have a feeling that maybe https://github.com/NuGet/Home/issues/11037 is related, but I'm not really sure. However, adding such a test is outside the scope for this PR. I'm just saying it as a FYI.

Reply:
I think it would be a good idea to call base methods and perform content validation in these tests.
I don't think that such a simple test is going to detect problems described in NuGet/Home#11037, but at least it will give some basic level of confidence that the actual file writing code works on all tested platforms (without deferring to higher-level tests).